doc/design/eventdriven-callbacks.md

```diff
@@ -0,0 +1,215 @@
+## Problem Description
+
+We require a public API so that a developer can register callbacks for certain
+events like the arrival of a sample.
+
+## Name of the design element
+
+ - ReactAl = React And Listen
+ - Reactor
+ - Eventler
+ - LiAR - Listen And React
+
+
+## Terminology
+
+ - **event driven** a one time reaction which is caused directly by an event.
+      Example: a new sample has been delivered to a subscriber.
+ - **state driven** a repeating reaction which is caused by a specific structural 
+      state until the state changes.
+      Example: a subscriber has stored samples which were not inspected by the user.
+ - **robust API** we use this term to describe an API which is nearly impossible 
```

Comment:
I guess we should avoid the term "nearly". This can be misunderstood. What about "Keep the misuse potential to a minimum"?